lib/logic/player/content.dart

```diff
@@ -117,17 +117,87 @@ class ContentMap<V> {
     );
     _map[key ?? T] = value;
   }
+
+  /// Look up the value of [key], or add a new entry if it isn't there.
+  ///
+  /// If [key] was explicitly provided, will use it instead.
+  V putIfAbsent<T extends Content>(V Function() ifAbsent, {Type? key}) {
+    assert(
+      Content.enumerate().contains(key ?? T),
+      "Specified type must be a subtype of Content",
+    );
+    return _map.putIfAbsent(key ?? T, ifAbsent);
+  }
+
+  /// Removes all entries from the map.
+  void clear() {
+    _map.clear();
+  }
+}
+
+/// A container for list of all content types.
+///
+/// This is like a [ContentMap] that contains lists and
+/// always guarantees to have a value in it for given content type.
+class ContentTuple {
+  final _map = ContentMap<List<Content>>();
```

Comment:
This would not be easier, because:
* we would need to write `contentPick` to implement `get`
* we could forget to add new value in `merged` getter
* `any` method, `empty` and `notEmpty` getters would look especially weird
* ...same for any other collection method that we could add here, since we can iterate over the map, and we can't do this with the way you suggest to implement it

> Instead get a compile time guarantee that a the list of songs exists.

We guarantee this in the constructor that we initalized all the types, which is acceptable here.